lib/rubocop/cop/rails/where_missing.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Use `where.missing(...)` to find missing relationship records.
+      #
+      # This cop is enabled in Rails 6.1 or higher.
+      #
+      # @example
+      #   # bad
+      #   Post.left_joins(:author).where(authors: { id: nil })
+      #
+      #   # good
+      #   Post.where.missing(:author)
+      #
+      class WhereMissing < Base
+        include RangeHelp
+        extend AutoCorrector
+        extend TargetRailsVersion
+
+        MSG = 'Use `where.missing(:%<left_joins_association>s)` instead of ' \
+              '`%<left_joins_method>s(:%<left_joins_association>s).where(%<where_association>s: { id: nil })`.'
+        RESTRICT_ON_SEND = %i[left_joins left_outer_joins].freeze
+
+        minimum_target_rails_version 6.1
+
+        # @!method where_node_and_argument(node)
+        def_node_search :where_node_and_argument, <<~PATTERN
+          $(send ... :where (hash <(pair $(sym _) (hash (pair (sym :id) (nil))))...> ))
+        PATTERN
+
+        # @!method missing_relationship(node)
+        def_node_search :missing_relationship, <<~PATTERN
+          (pair (sym _) (hash (pair (sym :id) (nil))))
+        PATTERN
+
+        def on_send(node)
+          return unless node.first_argument.sym_type?
+
+          where_node_and_argument(root_receiver(node)) do |where_node, where_argument|
+            next unless same_relationship?(where_argument, node.first_argument)
+
+            range = range_between(node.loc.selector.begin_pos, node.loc.expression.end_pos)
+            register_offense(node, where_node, where_argument, range)
+            break
+          end
+        end
+
+        private
+
+        def root_receiver(node)
+          node&.parent&.send_type? ? root_receiver(node.parent) : node
+        end
+
+        def same_relationship?(where, left_joins)
+          where.value.to_s.match?(/#{left_joins.value}s?/)
```

Comment:
I believe you have to constrain the regex (with \A and \z) and otherwise you are checking whether left_join is contained within the where, e.g. `"foobarcontainer".match?(/foos?/) => true`

Reply:
Thank you so much! I fixed it and added test https://github.com/rubocop/rubocop-rails/blob/851b927959623b551fb5384f84bf6cde7d24ae6d/spec/rubocop/cop/rails/where_missing_spec.rb#L263-L264